### PR TITLE
Functional test cases for Mutations and Queries via Guest user

### PR DESCRIPTION
**Pull request by jineshparakh**
CBG-0000

Describe your PR here...
- Added Functional test cases for Mutations and Queries via Guest user

## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## Dependencies (if applicable)
- [ ] Link upstream PRs
- [ ] Update Go module dependencies when merged

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [ ] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/000/

### CODE REVIEW

File: rest/functionsapitest/graphql_test.go [hunk 1]
```diff
@@ -484,3 +484,79 @@ func TestInvalidGraphQLConfigurationValues(t *testing.T) {
...
+func TestGraphQLMutationsGuest(t *testing.T) {
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{GuestEnabled: true, EnableUserQueries: true})
+	if rt == nil {
+		return
+	}
+	defer rt.Close()
+	rt.DatabaseConfig = &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+		GraphQL:       &kTestGraphQLConfig,
+		UserFunctions: &kTestGraphQLUserFunctionsConfig,
+	},
```
Comment (moshaad7) [inline rest/functionsapitest/graphql_test.go]: is this some go fmt fault ? 
},
}

File: rest/functionsapitest/graphql_test.go [hunk 2]
```diff
@@ -484,3 +484,79 @@ func TestInvalidGraphQLConfigurationValues(t *testing.T) {
...
+func TestGraphQLQueriesGuest(t *testing.T) {
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{GuestEnabled: true, EnableUserQueries: true})
+	if rt == nil {
+		return
+	}
+	defer rt.Close()
+	rt.DatabaseConfig = &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+		GraphQL:       &kTestGraphQLConfig,
+		UserFunctions: &kTestGraphQLUserFunctionsConfig,
+	},
```
Comment (moshaad7) [inline rest/functionsapitest/graphql_test.go]: },
}

**Review by moshaad7 — COMMENTED** [review]
minor nits

File: rest/functionsapitest/graphql_test.go (hunk 1 above)
Comment (jineshparakh) [reply rest/functionsapitest/graphql_test.go]: Yep, go fmt automatically formats it this way. Can't really do anything about it.

**Review by toad-sage — APPROVED** [review]

**Pull request merged**